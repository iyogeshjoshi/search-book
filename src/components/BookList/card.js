import React from "react";

const Card = ({ key, book }) => {
  return (
    <div key={`selected-book-${key}`} className="book-card">
      <div className="title" title={book.title}>
        {book.title}
      </div>
      <div className="author">By {book.author.author}</div>
      <div className="summary" title={book.summary}>
        {book.summary}
      </div>
    </div>
  );
};

export default Card;
