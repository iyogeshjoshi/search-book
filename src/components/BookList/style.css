.search-result {
  margin-top: 10vmin;
  padding: 15px;
}
/* .search-result:after {
  content: "";
  display: table;
  clear: both;
} */
.book-card {
  background-color: antiquewhite;
  color: black;
  /* height: 40vmin; */
  max-height: 50vmin;
  float: left;
  overflow: hidden;
  margin: 5px;
  min-width: 250px;
  width: 50vmin;
  padding: 20px 10px;
  text-align: left;
}

@media screen and (min-device-width: 1024px) and (max-device-width: 1600px) {
  .book-card {
    min-width: 250px;
    width: 45vmin;
    margin-left: 8vmin;
  }
}

@media screen and (min-width: 768px) and (max-width: 1024px) {
  .book-card {
    max-width: 100vmin;
    width: 100vmin;
    margin-left: 10vmin;
  }
}

@media screen and (max-width: 768px) {
  .book-card {
    max-width: 90vmin;
    width: 90vmin;
  }
}

.book-card .title {
  font-weight: bold;
  padding: 0 5px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.book-card .author {
  font-weight: lighter;
  text-align: right;
  padding: 0 10px;
  margin-top: 10px;
  margin-bottom: 15px;
  font-size: medium;
}

.book-card .summary {
  display: -webkit-box;
  box-orient: vertical;
  -webkit-box-orient: vertical;
  line-clamp: 3;
  -webkit-line-clamp: 3;
  overflow: hidden;
  text-align: left;
  font-style: italic;
  font-size: 16px;
}
