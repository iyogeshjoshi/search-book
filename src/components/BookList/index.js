import React from "react";
import Card from "./card";
import "./style.css";

const BookList = ({ books }) => {
  return (
    <>
      <section className="search-result">
        {books.size > 0 &&
          [...books.values()].map((book, index) => (
            <Card key={`CARD-${index}`} index={index} book={book} />
          ))}
      </section>
    </>
  );
};

export default BookList;
