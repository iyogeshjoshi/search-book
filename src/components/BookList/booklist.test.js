import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import BookList from "./";

describe("Renders BookList correctly", () => {
  it("matches snapshot", () => {
    const Books = [];
    const bookList = renderer.create(<BookList books={Books} />).toJSON();

    expect(bookList).toMatchSnapshot();
  });
  it("renders all the passed books", () => {
    const Books = new Map([
      [
        1,
        {
          title: "title 1",
          author: {
            id: 1,
            author: "author 1",
          },
          summary: "summary 1",
        },
      ],
      [
        2,
        {
          title: "title 2",
          author: {
            id: 2,
            author: "author 2",
          },
          summary: "summary 2",
        },
      ],
    ]);
    const { getByText } = render(<BookList books={Books} />);

    expect(getByText(/title 1/i)).toBeTruthy();
    expect(getByText(/author 1/i)).toBeTruthy();
    expect(getByText(/summary 1/i)).toBeTruthy();
    expect(getByText(/title 2/i)).toBeTruthy();
    expect(getByText(/author 2/i)).toBeTruthy();
    expect(getByText(/summary 2/i)).toBeTruthy();
  });
});
