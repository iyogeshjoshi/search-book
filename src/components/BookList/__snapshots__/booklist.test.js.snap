// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders BookList correctly matches snapshot 1`] = `
<section
  className="search-result"
/>
`;
